packages/nexus/src/context.ts

```diff
@@ -0,0 +1,131 @@
+  },
+  trace(message: string, meta?: LogMetadata): any {
+    return getLogger().trace(message, meta);
+  },
+  debug(message: string, meta?: LogMetadata): any {
+    return getLogger().debug(message, meta);
+  },
+  info(message: string, meta?: LogMetadata): any {
+    return getLogger().info(message, meta);
+  },
+  warn(message: string, meta?: LogMetadata): any {
+    return getLogger().warn(message, meta);
+  },
+  error(message: string, meta?: LogMetadata): any {
+    return getLogger().error(message, meta);
+  },
+};
+
+// TODO: also support getting a metrics handler.
+
+/**
+ * Returns a client to be used in a Nexus Operation's context, this Client is powered by the same Connection that the
+ * worker was created with.
+ */
+export function getClient(): Client {
+  return getHandlerContext<HandlerContext>().client;
+}
+
+export interface WorkflowHandle<_T> {
+  readonly workflowId: string;
+  readonly runId: string;
+}
+
+export async function startWorkflow<T extends Workflow>(
+  workflowTypeOrFunc: string | T,
+  workflowOptions: WorkflowStartOptions<T>,
+  nexusOptions: nexus.StartOperationOptions
+): Promise<WorkflowHandle<T>> {
+  const links = Array<temporal.api.common.v1.ILink>();
+  if (nexusOptions.links?.length > 0) {
+    for (const l of nexusOptions.links) {
+      try {
+        links.push({
+          workflowEvent: convertNexusLinkToWorkflowEventLink(l),
+        });
+      } catch (error) {
+        log.warn('failed to convert Nexus link to Workflow event link', { error });
+      }
+    }
+  }
+  const internalOptions: InternalWorkflowStartOptions = { links, requestId: nexusOptions.requestId };
+
+  if (workflowOptions.workflowIdConflictPolicy === 'USE_EXISTING') {
+    internalOptions.onConflictOptions = {
+      attachLinks: true,
+      attachCompletionCallbacks: true,
+      attachRequestId: true,
+    };
+  }
+
+  if (nexusOptions.callbackURL) {
+    internalOptions.completionCallbacks = [
+      {
+        nexus: { url: nexusOptions.callbackURL, header: nexusOptions.callbackHeaders },
+        links, // pass in links here as well, the server dedupes them.
+      },
+    ];
+  }
+  (workflowOptions as any)[InternalWorkflowStartOptionsKey] = internalOptions;
+  const handle = await getClient().workflow.start<T>(workflowTypeOrFunc, workflowOptions);
+  if (internalOptions.backLink?.workflowEvent != null) {
+    try {
+      handlerLinks().push(convertWorkflowEventLinkToNexusLink(internalOptions.backLink.workflowEvent));
+    } catch (error) {
+      log.warn('failed to convert Workflow event link to Nexus link', { error });
+    }
+  }
+  return { workflowId: handle.workflowId, runId: handle.firstExecutionRunId };
+}
+
+export type WorkflowRunOperationHandler<I, O> = (
+  input: I,
+  options: nexus.StartOperationOptions
+) => Promise<WorkflowHandle<O>>;
+
+export class WorkflowRunOperation<I, O> implements nexus.OperationHandler<I, O> {
+  constructor(readonly handler: WorkflowRunOperationHandler<I, O>) {}
```

Comment:
Given that `WorkflowRunOperation` implements `OperationHandler`, isn't there an argument that `WorkflowRunOperation` should be called `WorkflowRunOperationHandler` and that the function passed into the constructor (type currently called `WorkflowRunOperationHandler`) should be called `startHandler: WorkflowRunOperationStartHandler`?